### DISCUSSION TITLE
vSphere CSI driver

### DISCUSSION THREAD

**Discussion by LennertMertens** [okd-project/okd#1804] [Q&A]
Hi does anybody have some experience iwth UPI provisioned OKD clusters on VMware vSphere? I'm currently deploying 4.13 but I seem to have som eissues with configuring the CSI driver. Went through the documentation and did everything explained there. Still seeing some issues however.

Thanks!

**Comment by PolarisPT** [comment]
Hi i'm also having issues i cannot provision PVC's.
"
failed to provision volume with StorageClass "thin-csi": rpc error: code = Internal desc = failed to create volume. Error: auth service: no shared datastore found for block volume provisioning
"

**Reply by jcpowermac** [reply]
Can you provide a must gather? Do you have a shared datastore? How did you install, UPI or IPI?

**Comment by PolarisPT** [comment]
IPI latest version 4.14.0-0.okd-2024-01-06-084517
Yes i have a shared datastore.

some more info:

```
[root@bastion install_11012024_1_SUCCESS]# oc get storageclasses thin-csi
NAME                 PROVISIONER              RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
thin-csi (default)   csi.vsphere.vmware.com   Delete          WaitForFirstConsumer   true                   24h
[root@bastion install_11012024_1_SUCCESS]# oc describe storageclasses thin-csi
Name:                  thin-csi
IsDefaultClass:        Yes
Annotations:           storageclass.kubernetes.io/is-default-class=true
Provisioner:           csi.vsphere.vmware.com
Parameters:            StoragePolicyName=openshift-storage-policy-gto-nprd-27gdq
AllowVolumeExpansion:  True
MountOptions:          <none>
ReclaimPolicy:         Delete
VolumeBindingMode:     WaitForFirstConsumer
Events:                <none>
```

Also check openshift-storage-policy-gto-nprd-27gdq in Vcenter and seems to be all ok there....

**Reply by jcpowermac** [reply]
Must gather please

**Reply by PolarisPT** [reply]
Here it is -> https://drive.google.com/file/d/1l29kkRg5pdnc9yp8gcABgfQkenCtv4Lo/view?usp=drive_link

**Reply by jcpowermac** [reply]
@PolarisPT I requested access

**Reply by PolarisPT** [reply]
@jcpowermac shared.

**Reply by jcpowermac** [reply]
```
pods/cluster-storage-operator-5c4d695bb5-b47mq/cluster-storage-operator/cluster-storage-operator/logs/current.log:233:2024-01-15T08:46:02.608607452Z I0115 08:46:02.608426       1 status_controller.go:213] clusteroperator/storage diff {"status":{"conditions":[{"lastTransitionTime":"2024-01-12T16:44:09Z","message":"VSph
ereCSIDriverOperatorCRDegraded: VMwareVSphereControllerDegraded: unable to fetch default datastore url: failed to access datastore /Snt/datastore/AAA_SNT_TOOLS_T0/AAA_SNT_TOOLS_T0_DS01: datastore '/Snt/datastore/AAA_SNT_TOOLS_T0/AAA_SNT_TOOLS_T0_DS01' not found","reason":"VSphereCSIDriverOperatorCR_VMwareVSphereContro
ller_SyncError","status":"True","type":"Degraded"},{"lastTransitionTime":"2024-01-12T15:23:10Z","message":"VSphereCSIDriverOperatorCRProgressing: All is well","reason":"AsExpected","status":"False","type":"Progressing"},{"lastTransitionTime":"2024-01-12T15:22:52Z","message":"DefaultStorageClassControllerAvailable: Sto
rageClass provided by supplied CSI Driver instead of the cluster-storage-operator\nVSphereCSIDriverOperatorCRAvailable: All is well\nVSphereProblemDetectorMonitoringControllerAvailable: vsphere-problem-detector alerts are enabled\nVSphereProblemDetectorControllerAvailable: [defaultDatastore \"/Snt/datastore/AAA_SNT_TO
OLS_T0/AAA_SNT_TOOLS_T0_DS01\" in vSphere configuration: couldn't find Datastore named \"/Snt/datastore/AAA_SNT_TOOLS_T0/AAA_SNT_TOOLS_T0_DS01\" in datacenter Snt, failed to access datastore /Snt/datastore/AAA_SNT_TOOLS_T0/AAA_SNT_TOOLS_T0_DS01: cannot find datastore /Snt/datastore/AAA_SNT_TOOLS_T0/AAA_SNT_TOOLS_T0_DS
01 in datacenter Snt, cannot find datastore /Snt/datastore/AAA_SNT_TOOLS_T0/AAA_SNT_TOOLS_T0_DS01 in datacenter Snt]","reason":"AsExpected","status":"True","type":"Available"},{"lastTransitionTime":"2024-01-11T17:37:02Z","message":"VSphereCSIDriverOperatorCRUpgradeable: All is well","reason":"AsExpected","status":"Tru
e","type":"Upgradeable"}]}}    
```

**Reply by jcpowermac** [reply]
btw datastore clusters are unsupported if storage drs is enabled
Storage VMotion causes dataloss

**Reply by jcpowermac** [reply]
@PolarisPT this is a little odd. I see the vsphere problem detector has no problem accessing that datastore but the drivers do.

Are you sure each ESXi host where these virtual machines are running has access to that datastore?

**Reply by PolarisPT** [reply]
@jcpowermac well if the IPI did not have any issues provisioning the nodes for the cluster in the same datastore why would the cluster after installation have any issues?
Where can i double check the credentials used by the driver?

Thanks

**Reply by PolarisPT** [reply]
@jcpowermac maybe i gave the wrong datastore option in install-config.yaml...
AAA_SNT_TOOLS_T0 is the datastore cluster 
AAA_SNT_TOOLS_T0_DS01 is a member 

In install-config.yaml i had :
AAA_SNT_TOOLS_T0_DS01 

Thanks

**Reply by jcpowermac** [reply]
`/<datacenter>/datastore/<ds_cluster_name>/<ds_name>`

**Reply by jcpowermac** [reply]
If you provided the wrong credentials cluster wouldn't have installed

**Reply by jcpowermac** [reply]
>  well if the IPI did not have any issues provisioning the nodes for the cluster in the same datastore why would the cluster after installation have any issues?

vmotion
but since I don't know your vsphere design I am just guessing to try to determine a cause.

**Comment by gnufied** [comment]
Also, ensure that you specify full username@domain in vsphere credentials.

**Comment by PolarisPT** [comment]
Fixed my issue !
It was related with the vmware user, i had to use "DOMAIN\USER" in the credentials secrets instead of "USER" only.
